store/src/java/com/zimbra/cs/service/admin/AddDistributionListMember.java

```diff
@@ -63,6 +65,13 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
             throw ServiceException.INVALID_REQUEST("members to add not specified", null);
         }
 
+        for (String member : memberList) {
```

Comment:
Rather than adding this logic here we should check the method prov.addGroupMembers(group, members); as this is responsible for adding DLs. So we should check what's breaking or need to improve logic.

Reply:
Yes, I understand the logic is written inside the method prov.addGroupMembers(group, members);
It is breaking for not validating account before adding it to the DL.
My motive is to write this here is that we will be checking whether the account is valid or not beforehand so that we could save method calling when it is not required at all and throw appropriate exception to user. Also, inside addGroupMembers() we need to write this throw exception code at 2 places, one for addDynamicGroupMembers() and another for addDistributionListMembers().
Please let me know if, I should do this inside addGroupMembers(). Thanks!

Reply:
How often is this condition being true where we get account as null?
If it is often happening then it makes sense to do as below but if it is not happening often then we can add it inside both the methods as then we will not need to process list of members everytime.

Reply:
It is unpredictable, could be frequent or could not happen for a long time. As input is taken from user so can't determine its occurrence.
So now writing it inside both of the methods as suggested.